galley/pkg/config/analysis/analyzers/analyzers_test.go

```diff
@@ -88,6 +89,66 @@ var testGrid = []testCase{
 			{msg.PodMissingProxy, "Pod/default/noninjectedpod"},
 		},
 	},
+	{
+		name: "gatewayNoWorkload",
```

Comment:
Do you think it would make sense to combine any of these test cases? e.g. how we're doing it with tests against the virtualservice analyzers.

Reply:
I am uncomfortable with the test comparing the path to the failing Resource and the template string.  I'd prefer if the test worked against the exact message.

The Gateway port check needs to check many similar things.  I am expecting output similar to:

```
... not exposed on the workload (pod selector custom=bar; port 80)
... not exposed on the workload (pod selector custom=foo; port 8001)
... not exposed on the workload (pod selector istio=ingressgateway; port 8004)
```

Because we only check the templates I there is no test that I am supply the correct args.
Because there are so many extremely similar cases I was getting confused so I created a bunch of variations and gave each one its own test.

If you can give me a way for `expected` to show the complete output I could combine some of my test files.  Not all of them, because I need separate tests for explicit system ingressgateway and unsupplied system ingressgateway.

Reply:
Thank you, that's really useful feedback about the limitations of the current test framework.

The way I've approached this so far is to have unique resource names (in this case, gateway names) for each expected output message, rather than one resource that needs to generate multiple messages of the same type but different inputs. I'm not going to claim that's the right or best or only way. :) Perhaps we could extend this a bit so that you can optionally specify the expected message inputs in the `expected` section.